src/charts/time_series_labeler.coffee

```diff
@@ -56,112 +85,156 @@ Ember.Charts.TimeSeriesLabeler = Ember.Mixin.create
+      d3.time.years(start, stop).length
 
-    # if we have too many quarters then we only display quarter labels
-    # on years
-    if quarters.length > @get('maxNumberOfLabels')
-      @labelledYears(start, stop)
-    else
-      quarters
+  # The labeller type used to create the labels for each domain type
+  # Note that quarters uses a month labeller to create the labels
+  domainTypeToLabellerType =
+    'S': 'seconds'
+    'H': 'hours'
+    'D': 'days'
+    'W': 'weeks'
+    'M': 'months'
+    'Q': 'months'
+    'Y': 'years'
 
-  monthsBetween: (start, stop, skip = 1) ->
-    d3.time.months(start, stop).filter((d, i) -> !(i % skip))
+  # The lengthened representation for each domain type. This is different from
+  # doaminTypeToLabellerType
+  domainTypeToLongDomainType =
+    'S': 'seconds'
+    'H': 'hours'
+    'D': 'days'
+    'W': 'weeks'
+    'M': 'months'
+    'Q': 'quarters'
+    'Y': 'years'
 
-  # the months which should be labelled
-  labelledMonths: (start, stop) ->
-    months = @monthsBetween(start, stop)
+  longDomainTypeToDomainType =
+    'seconds': 'S'
+    'hours': 'H'
+    'days': 'D'
+    'weeks': 'W'
+    'months': 'M'
+    'quarters': 'Q'
+    'years': 'Y'
 
-    # if we have too many months then we reduce to the maximum number of
-    # months labelled such that they are uniformly distributed in the
-    # range
-    if months.length > @get('maxNumberOfLabels')
-      skipVal = Math.ceil(months.length / @get('maxNumberOfLabels'))
-      @monthsBetween(start, stop, skipVal)
-    else
-      months
+  # Dynamic x labelling tries to intelligently limit the number of labels
+  # along the x axis to a specified limit. In order to do this, time
+  # specificity is callibrated (e.g. for a range of 2 years, instead of having
+  # the labels be in years, the specificity is increased to quarters)
+  # If the minTimeSpecificity or maxTimeSpecificity are set, then the labels
+  # are limited to fall between the time units between these bounds.
+  dynamicXLabelling: (start,stop) ->
+    ind1 = @get('DOMAIN_ORDERING').indexOf(@get 'minTimeSpecificity')
+    ind2 = @get('DOMAIN_ORDERING').indexOf(@get 'maxTimeSpecificity')
+    
+    # Refers to the metrics used for the labelling
+    domainTypes = @get('DOMAIN_ORDERING')[ind1..ind2]
 
-  weeksBetween: (start, stop, skip = 1) ->
-    d3.time.weeks(start, stop).filter((d, i) -> !(i % skip))
+    # The labeller type to create the labels for each metric
+    labellerTypes =
+      domainTypeToLabellerType[domainType] for domainType in domainTypes
 
-  secondsBetween: (start, stop, skip = 1) ->
-    d3.time.seconds(start, stop).filter((d, i) -> !(i % skip))
+    #The time span in various metrics
+    times = (numTimeBetween[domainTypeToLongDomainType[d]](start, stop) for d in domainTypes)
 
-  # the weeks which should be labelled
-  labelledWeeks: (start, stop) ->
-    weeks = @weeksBetween(start, stop)
+    labels = null
+    xLabels = @get 'maxNumberOfLabels'
```

Comment:
could you delete this and just use `@get 'maxNumberOfLabels` in line? changing the variable name on me is confusing